### PR TITLE
Fixes broken conversion between EmailNotification and string

### PR DESCRIPTION
**Pull request by nicmr**
Just using the built-in go type conversions often does not play well with fields omitted due to `omitempty` and their complete lack of producing errors on invalid values. This PR makes the conversions more explicit and allows error handling, if a type variant is not allowed.

This fixes `jelease newreleases apply` of a resource configured with emailNotification = "none" to immediately result in a divergence.

To do before ready:
- [ ] add tests

### CODE REVIEW

File: pkg/newreleases/newreleases.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func ExclusionSliceFromCfg(exclusions []config.NewReleasesExclusion) []newreleas
 func ProjectToCfg(project newreleases.Project) config.NewReleasesProject {
 	return config.NewReleasesProject{
 		Name:               project.Name,
-		EmailNotification:  string(project.EmailNotification),
+		EmailNotification:  emailNotificationToString(project.EmailNotification),
```
Comment (jilleJr) [inline pkg/newreleases/newreleases.go:155]: Suggest creating a new type in pkg/config, and take inspiration from `pkg/config/loglevel.go` on parsing.

Then move the `emailNotificationFromString` function into there as `ParseEmailNotification`

And move `emailNotificationToString` to be a `ReleaseEmailNotification.String()` method on the type.

File: pkg/newreleases/newreleases.go [hunk 2]
```diff
@@ -246,19 +246,65 @@ func (nr NewReleases) ApplyLocalConfig(options ApplyLocalConfigOptions) error {
...
+		s = "daily"
+	case newreleases.EmailNotificationWeekly:
+		s = "weekly"
+	case newreleases.EmailNotificationDefault:
+		s = "default"
+	default:
+		s = "none"
+	}
+	return s
+}
```
Comment (jilleJr) [inline pkg/newreleases/newreleases.go:310]: Think it's sufficient to just do `string(notification)` here. No need to do "validation" stuff twice / ensuring it's a valid value. Sufficient to only do it during parsing

File: pkg/newreleases/newreleases.go [hunk 3]
```diff
@@ -246,19 +246,65 @@ func (nr NewReleases) ApplyLocalConfig(options ApplyLocalConfigOptions) error {
...
+	case newreleases.EmailNotificationDaily:
+		s = "daily"
+	case newreleases.EmailNotificationWeekly:
+		s = "weekly"
+	case newreleases.EmailNotificationDefault:
+		s = "default"
+	default:
+		s = "none"
+	}
+	return s
```
Comment (jilleJr) [inline pkg/newreleases/newreleases.go:309]: Suggest to try upstream these changes to their repo <https://github.com/newreleasesio/client-go>, to add `.String() string` and a `ParseEmailNotification(string) (EmailNotification, error)` function there. No clue if they would accept it, but could be a nice thing to do.

**Comment by applejag** [comment]
Gonna close this as "discontinued"

**Pull request closed without merging**